### DISCUSSION TITLE
Error when trying to list all the available casks

### DISCUSSION THREAD

**Discussion by nibin-matteo-git** [Homebrew/discussions#1303] [Casks] [answered]
```
nibin@Nibins-Air ~ % brew search --casks
Usage: brew search, -S [options] text|/regex/ [...]

Perform a substring search of cask tokens and formula names for text. If
text is flanked by slashes, it is interpreted as a regular expression. The
search for text is extended online to homebrew/core and homebrew/cask.

      --formula, --formulae        Search online and locally for formulae.
      --cask, --casks              Search online and locally for casks.
      --desc                       Search for formulae with a description
                                   matching text and casks with a name
                                   matching text.
      --pull-request               Search for GitHub pull requests containing
                                   text.
      --open                       Search for only open GitHub pull requests.
      --closed                     Search for only closed GitHub pull
                                   requests.
      --macports                   Search for text in the given package
                                   manager's list.
      --fink                       Search for text in the given package
                                   manager's list.
      --opensuse                   Search for text in the given package
                                   manager's list.
      --fedora                     Search for text in the given package
                                   manager's list.
      --debian                     Search for text in the given package
                                   manager's list.
      --ubuntu                     Search for text in the given package
                                   manager's list.
  -d, --debug                      Display any debugging information.
  -q, --quiet                      Make some output more quiet.
  -v, --verbose                    Make some output more verbose.
  -h, --help                       Show this message.

Error: Invalid usage: This command requires at least 1 text or regex argument.
nibin@Nibins-Air ~ % 
```

my brew is up to date as you can see from the following output,

```
nibin@Nibins-Air ~ % brew update --casks
Already up-to-date.
```

and when i give brew outdated i don't get anything (I assume that's because everything is up to date)

```
nibin@Nibins-Air ~ % brew outdated
nibin@Nibins-Air ~ % 
```

This is the output that I get when i give brew doctor

```
nibin@Nibins-Air ~ % brew doctor
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: "config" scripts exist outside your system or Homebrew directories.
`./configure` scripts often look for *-config scripts to determine if
software packages are installed, and which additional flags to use when
compiling and linking.

Having additional scripts in your path can confuse software installed via
Homebrew if the config script overrides a system or Homebrew-provided
script of the same name. We found the following "config" scripts:
  /usr/local/bin/python3.9-config
  /usr/local/bin/python3-config
  /Library/Frameworks/Python.framework/Versions/3.9/bin/python3.9-config
  /Library/Frameworks/Python.framework/Versions/3.9/bin/python3-config
nibin@Nibins-Air ~ % 
```

**Comment by Rylan12** [comment, marked as answer]
> ```
> Error: Invalid usage: This command requires at least 1 text or regex argument.
> ```

You're getting an error because the command you're running is incomplete. You're running a search command but you're not actually searching for anything.

Try

```sh
brew search --cask iterm2
```

Obviously, replace `iterm2` which whatever cask you're looking for.

**Reply by nibin-matteo-git** [reply]
So how do I get the list of everything that's available
because you're supposed to get the list of all casks when you enter brew search --casks right

**Reply by Rylan12** [reply]
`brew casks` lists all casks

**Reply by tlk** [reply]
> because you're supposed to get the list of all casks when you enter `brew search --casks` right

It looks like this command used to work but was deprecated before it was finally removed nine days ago in this large commit https://github.com/Homebrew/brew/commit/d73351251cb90fcc353557e6dfd4497a6ad83aef#diff-53df9e16c5fa45dd859ce5af400a448876e8476344a1d66547cb4567b2e8bddd

The commit message is rather generic and does not provide any reasoning for this change, but the PR https://github.com/Homebrew/brew/pull/11075 was labeled "critical" so I guess it was deemed important to merge the PR in a hurry.

**Reply by Rylan12** [reply]
Ugh, yeah it looks like it wasn't properly deprecated 😭 

It should probably be re-added and properly deprecated. Either way, though, you should use `brew casks` not `brew search --cask` because if it's readded it will be gone for good soon.

https://github.com/Homebrew/brew/pull/11182

**Reply by tlk** [reply]
Hi Rylan, sorry if that came out wrong. I was looking at what that PR did and why and failed to find a roadmap or similar explaining the reasoning behind it. I don't have any beef with this particular command being deprecated (properly or not), please don't break your back on this.

**Reply by Rylan12** [reply]
No worries, it should probably be done anyway as it has the potential to break lots of stuff (although I'll be honest I'm a little surprised that this is the first report that I've seen. Maybe it's not as widely used as I assumed it might be).

We don't really have a roadmap where we lay these things out. Generally, things are deprecated because they have been replaced with a newer (and hopefully better) alternative or they are something that we are choosing to no longer support (which could be due to a variety of reasons). In this case, the reason it was deprecated was that it doesn't really fit in with a "search" command and we have a newer (and much faster) `brew casks` command that does the same thing.

**Reply by nibin-matteo-git** [reply]
It works 
thanks guys

**Reply by nibin-matteo-git** [reply]
correct me if I'm wrong 
Do we use brew formulae to list all the formulae

**Reply by carlocab** [reply]
You are correct.
```
❯ brew help formulae
Usage: brew formulae

List all locally installable formulae including short names.
```
